src/explore-education-statistics-common/src/components/ContentHtml.tsx

```diff
@@ -1,31 +1,107 @@
 import sanitizeHtml, { SanitizeHtmlOptions } from '@common/utils/sanitizeHtml';
+import parseHtmlString, { DOMNode, domToReact } from 'html-react-parser';
+import { Element } from 'domhandler/lib/node';
 import classNames from 'classnames';
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
+import { GlossaryEntry } from '@common/services/types/glossary';
+import Modal from '@common/components/Modal';
+import InfoIconButton from '@common/components/InfoIconButton';
+import useMounted from '@common/hooks/useMounted';
+import ButtonText from '@common/components/ButtonText';
 
 export interface ContentHtmlProps {
   className?: string;
   html: string;
   sanitizeOptions?: SanitizeHtmlOptions;
   testId?: string;
+  getGlossaryEntry?: (slug: string) => Promise<GlossaryEntry>;
 }
 
 const ContentHtml = ({
   className,
   html,
   sanitizeOptions,
   testId,
+  getGlossaryEntry,
 }: ContentHtmlProps) => {
+  const { isMounted } = useMounted();
+
+  const [glossaryEntry, setGlossaryEntry] = useState<GlossaryEntry | undefined>(
+    undefined,
+  );
+
   const cleanHtml = useMemo(() => {
     return sanitizeHtml(html, sanitizeOptions);
   }, [html, sanitizeOptions]);
 
+  const parsedContent = parseHtmlString(cleanHtml, {
+    replace: (node: DOMNode) => {
+      if (
+        getGlossaryEntry &&
+        node instanceof Element &&
+        node.name === 'a' &&
+        node.attribs &&
+        Object.keys(node.attribs).includes('data-glossary')
+      ) {
+        const linkText = domToReact(node.children);
+        const glossaryEntrySlug = node.attribs.href.split('#')[1];
+        return isMounted ? (
+          <>
+            <ButtonText
+              onClick={async () => {
+                const newGlossaryEntry = await getGlossaryEntry(
+                  glossaryEntrySlug,
+                );
+                setGlossaryEntry(newGlossaryEntry);
+              }}
+            >
+              {linkText}
+            </ButtonText>{' '}
+            <InfoIconButton
+              description={`Show glossary term definition for ${linkText}`}
+              testId={`glossary-info-icon-${glossaryEntrySlug}`}
+              onClick={async () => {
+                const newGlossaryEntry = await getGlossaryEntry(
+                  glossaryEntrySlug,
+                );
+                setGlossaryEntry(newGlossaryEntry);
+              }}
+            />
```

Comment:
Nice try but having two buttons next to each other is redundant and also going to be quite confusing for screenreaders as it would end up announcing the presence of two buttons.

We should simplify it to something like the following:

```jsx
<ButtonText testId={`glossary-entry-${glossaryEntrySlug}`} onClick={...}>
  {linkText}{' '}
  <InfoIcon description="(show glossary term definition)" />
</ButtonText>
```

Note the following key features:

- `InfoIconButton` should now simply be the appropriately styled `?` icon i.e `InfoIcon`. It should still be `aria-hidden`. The `description` prop is still used to provide the accessible text.
- There is only the single `ButtonText` with the `onClick` being used to show the modal.

If there is an ugly link underline that goes beneath both the link text and the icon, I would suggest we